src/tests/integration/connection/connectionV2Invite.spec.ts

```diff
@@ -0,0 +1,218 @@
+  }
+
+  let sourcePrivateKey = ecSecp256k1
+    .genKeyPair()
+    .getPrivate()
+    .toString('hex');
+
+  if (sourcePrivateKey.length !== 64) {
+    sourcePrivateKey =
+      '22f8f182a11f90e1a64d693c7cbf44ca017c50b6efc51445ea2f319210875c09';
+  }
+
+  // Generate random username
+  let username = `User${Math.random()
+    .toString(36)
+    .substring(7)}`;
+
+  let sourceUserWalletId: string;
+  let targetUserId: string;
+  let pSdk: PillarSdk;
+
+  const responseData = {
+    result: 'success',
+    message: 'Connection invitation was successfully sent',
+  };
+
+  const errInvalidWalletId = {
+    message: 'Could not find a Wallet ID to search by.',
+  };
+
+  const errInternal = {
+    message: 'Internal Server Error',
+  };
+
+  beforeAll(async () => {
+    pSdk = new PillarSdk({});
+    pSdk.configuration.setUsername('username');
+
+    if (env === 'test') {
+      const mockApi = nock('https://localhost:8080');
+      mockApi
+        .post('/register/keys')
+        .reply(200, {
+          expiresAt: '2015-03-21T05:41:32Z',
+          nonce: 'AxCDF23232',
+        })
+        .post('/register/auth')
+        .reply(200, {
+          authorizationCode: 'Authorization code',
+          expiresAt: '2011-06-14T04:12:36Z',
+        })
+        .post('/register/access')
+        .reply(200, {
+          accessToken: 'accessToken',
+          refreshToken: 'refreshToken',
+          walletId: 'targetWalletId',
+          userId: 'targetUserId',
+        })
+        .post('/register/keys')
+        .reply(200, {
+          expiresAt: '2015-03-21T05:41:32Z',
+          nonce: 'AxCDF23232',
+        })
+        .post('/register/auth')
+        .reply(200, {
+          authorizationCode: 'Authorization code',
+          expiresAt: '2011-06-14T04:12:36Z',
+        })
+        .post('/register/access')
+        .reply(200, {
+          accessToken: 'accessToken',
+          refreshToken: 'refreshToken',
+          walletId: 'sourceWalletId',
+          userId: 'sourceUserId',
+        })
+        .post('/connection/v2/invite')
+        .reply(200, responseData)
+        .post('/connection/v2/invite')
+        .reply(400, errInvalidWalletId)
+        .post('/connection/v2/invite')
+        .reply(500, errInternal)
+        .post('/connection/v2/invite');
+    }
+
+    try {
+      let walletRegister = {
+        privateKey: targetPrivateKey,
+        fcmToken: '987qwe1',
+        username,
+      };
+
+      let response = await pSdk.wallet.registerAuthServer(walletRegister);
+      targetUserId = response.data.userId;
+
+      username = `User${Math.random()
+        .toString(36)
+        .substring(7)}`;
+
+      walletRegister = {
+        privateKey: sourcePrivateKey,
+        fcmToken: '987qwe2',
+        username,
+      };
+
+      response = await pSdk.wallet.registerAuthServer(walletRegister);
+      sourceUserWalletId = response.data.walletId;
+    } catch (e) {
+      throw e;
```

Comment:
As you're only throwing an error inside that `catch` section, then you can safely remove the whole try..catch block and it would work the same, isn't it?